### PR TITLE
Migrate to GitHub Actions for CI

### PR DESCRIPTION
**Pull request by jacobwinch**
## What does this change?

This PR configures GitHub Actions for AMIgo (as a replacement for [TeamCity](https://teamcity.gutools.co.uk/admin/editBuildRunners.html?id=buildType:Tools_Amigo)).

## How to test

I've deployed a branch built via GitHub Actions to `CODE`.

## How can we measure success?

All configuration related to CI is now in version control.

## Have we considered potential risks?

Yes, CI performance may be slightly worse on GitHub Actions. As there are plans to address that over the next few months I'm not especially concerned about it.

### CODE REVIEW

File: cdk/script/ci [hunk 1]
```diff
@@ -2,27 +2,7 @@
 
 set -e
 
-nvm_available() {
```
Comment (jacobwinch) [inline cdk/script/ci:5]: All of this is now handled by `actions/setup-node@v2` / the GHA environment

File: project/plugins.sbt [hunk 2]
```diff
@@ -1,6 +1,6 @@
 addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.6.25")
 addSbtPlugin("com.typesafe.sbt" % "sbt-scalariform" % "1.3.0")
 addSbtPlugin("com.typesafe.sbt" % "sbt-native-packager" % "1.8.0")
-addSbtPlugin("com.gu" % "sbt-riffraff-artifact" % "1.1.12")
+addSbtPlugin("com.gu" % "sbt-riffraff-artifact" % "1.1.18")
```
Comment (jacobwinch) [inline project/plugins.sbt:4]: This is required as GitHub Actions support was added relatively recently

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      - uses: actions/checkout@v2
+      - uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ secrets.GU_RIFF_RAFF_ROLE_ARN }}
+          aws-region: eu-west-1
+      - uses: actions/setup-node@v2 # This is required by cdk
+        with:
+          cache: 'yarn'
+          cache-dependency-path: 'cdk/yarn.lock'
+          node-version-file: 'cdk/.nvmrc'
```
Comment (jacobwinch) [inline .github/workflows/ci.yaml:24]:
```
Run actions/setup-node@v2
Resolved cdk/.nvmrc as 14.17.1
Attempting to download 14.17.1...
```

https://github.com/guardian/amigo/blob/d66fc6d64830cf314c71839c42842f88958cd61f/cdk/.nvmrc#L1

🎉

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+      - main
+jobs:
+  CI:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - uses: actions/checkout@v2
+      - uses: aws-actions/configure-aws-credentials@v1
```
Comment (akash1810) [inline .github/workflows/ci.yaml:16]: Oooh! Shall we start to deprecate our version of this now then?

File: project/plugins.sbt (hunk 2 above)
Comment (akash1810) [reply project/plugins.sbt:4]: 👍🏽  #650 and its duplicate #656 can be closed too!

This version also means we can [remove](https://github.com/guardian/amigo/pull/656/files#diff-5634c415cd8c8504fdb973a3ed092300b43c4b8fc1e184f7249eb29a55511f91L33-L34) bucket references too as the plugin defaults them.

**Review by akash1810 — APPROVED** [review]

File: project/plugins.sbt (hunk 2 above)
Comment (jacobwinch) [reply project/plugins.sbt:4]: > 👍🏽 https://github.com/guardian/amigo/pull/650 and its duplicate https://github.com/guardian/amigo/pull/656 can be closed too!

Sorry, I didn't notice those (even though I seem to have approved one of them!) - third time lucky!

> This version also means we can [remove](https://github.com/guardian/amigo/pull/656/files#diff-5634c415cd8c8504fdb973a3ed092300b43c4b8fc1e184f7249eb29a55511f91L33-L34) bucket references too as the plugin defaults them.

Thanks for the suggestion 👍

Comment (akash1810) [reply project/plugins.sbt:4]: > This version also means we can [remove](https://github.com/guardian/amigo/pull/656/files#diff-5634c415cd8c8504fdb973a3ed092300b43c4b8fc1e184f7249eb29a55511f91L33-L34) bucket references too as the plugin defaults them.

Forgot to add the link! This change was added in https://github.com/guardian/sbt-riffraff-artifact/pull/63.

File: .github/workflows/ci.yaml (hunk 4 above)
Comment (jacobwinch) [reply .github/workflows/ci.yaml:16]: Yep, I tried to capture the list of tasks involved [here](https://trello.com/c/RFbp0hoi/1122-deprecate-guardian-actions-assume-aws-role). Please feel free to add anything that I've forgotten.

**Pull request merged**